airflow-core/src/airflow/ui/src/pages/TaskInstance/RenderedTemplates.tsx

```diff
@@ -16,14 +16,75 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import { Box, HStack, Table, Code } from "@chakra-ui/react";
+import { Box, HStack, Table } from "@chakra-ui/react";
 import { useParams } from "react-router-dom";
+import { PrismLight as SyntaxHighlighter } from "react-syntax-highlighter";
+import bash from "react-syntax-highlighter/dist/esm/languages/prism/bash";
+import json from "react-syntax-highlighter/dist/esm/languages/prism/json";
+import sql from "react-syntax-highlighter/dist/esm/languages/prism/sql";
+import yaml from "react-syntax-highlighter/dist/esm/languages/prism/yaml";
+import { oneLight, oneDark } from "react-syntax-highlighter/dist/esm/styles/prism";
 
 import { useTaskInstanceServiceGetMappedTaskInstance } from "openapi/queries";
 import { ClipboardRoot, ClipboardIconButton } from "src/components/ui";
+import { useColorMode } from "src/context/colorMode";
+
+SyntaxHighlighter.registerLanguage("json", json);
+SyntaxHighlighter.registerLanguage("yaml", yaml);
+SyntaxHighlighter.registerLanguage("sql", sql);
+SyntaxHighlighter.registerLanguage("bash", bash);
+
+const detectLanguage = (value: string): string => {
+  const trimmed = value.trim();
+
+  // Try to detect JSON
+  if (
+    (trimmed.startsWith("{") && trimmed.endsWith("}")) ||
+    (trimmed.startsWith("[") && trimmed.endsWith("]"))
+  ) {
+    try {
+      JSON.parse(trimmed);
+
+      return "json";
+    } catch {
+      // Not valid JSON, continue
+    }
+  }
+
+  // Try to detect YAML (basic heuristics)
+  if (trimmed.includes(":") && (trimmed.includes("\n") || trimmed.includes("- "))) {
```

Comment:
Detection safety:
- If a colon is matched it should be followed by a newline (`":\n"`)
- Should we not rather use a lib like https://www.npmjs.com/package/yaml to test parsing after some basic heuristics before making a decision (similar to JSON)?